ion/unmarshal.go

```diff
@@ -671,3 +701,48 @@ func indirect(v reflect.Value, wantPtr bool) reflect.Value {
 
 	return v
 }
+
+func (d *Decoder) decodeToStructWithAnnotation(v reflect.Value) error {
+	if err := d.attachAnnotations(v); err != nil {
+		return err
+	}
+
+	fields := fieldsFor(v.Type())
+	for _, field := range fields {
+		if field.name != AnnotationsTag {
+			field := findField(fields, field.name)
+			if field != nil {
+				subValue, err := findSubvalue(v, field)
+				if err != nil {
+					return err
+				}
+
+				if err := d.decodeTo(subValue); err != nil {
+					return err
+				}
+				break
+			}
+		}
+	}
+	return nil
+}
+
+func (d *Decoder) attachAnnotations(v reflect.Value) error {
+	fields := fieldsFor(v.Type())
+	field := findField(fields, AnnotationsTag)
+	if field != nil {
+		subValue, err := findSubvalue(v, field)
+		if err != nil {
+			return err
+		}
+		annotations := d.r.Annotations()
+		annotationsType := indirect(reflect.ValueOf(annotations), false).Type()
+		valueSlice := reflect.MakeSlice(annotationsType, 0, len(annotations))
+
+		for _, an := range annotations {
+			valueSlice = reflect.Append(valueSlice, reflect.ValueOf(an))
+		}
+		subValue.Set(valueSlice)
```

Comment:
Any reason we can't just do `subValue.Set(reflect.ValueOf(annotations))` here?

Reply:
The only reason was my lack of familiarity with Go!
Modified, thank.